module-services/service-desktop/endpoints/security/SecurityEndpointHelper.cpp

```diff
@@ -50,14 +50,26 @@ namespace sdesktop::endpoints
 
     auto SecurityEndpointHelper::processStatus(Context & /*context*/) -> http::Code
     {
+        auto result         = http::Code::NoContent;
         auto desktopService = dynamic_cast<ServiceDesktop *>(owner);
         auto security       = desktopService->getSecurity()->getEndpointSecurity();
 
-        if (security == EndpointSecurity::Allow) {
+        if (security.access == EndpointSecurity::Allow) {
             preventBlockingDevice();
+            result = http::Code::NoContent;
         }
-
-        return security == EndpointSecurity::Allow ? http::Code::NoContent : http::Code::Forbidden;
+        else {
+            switch (security.reason) {
+            case BlockReason::NoReason:
+            case BlockReason::DeviceLocked:
+                result = http::Code::Forbidden;
+                break;
+            case BlockReason::EulaNotAccepted:
+                result = http::Code::Locked;
```

Comment:
The code name for `EulaNotAccepted` block reason looks a bit confusing here, but I understand it is not easy to find a proper HTTP error code for our needs.

Reply:
@pawel-mudita I think I chose the most suitable code anyway from the available and unused ones